src/components/Todo.tsx

```diff
@@ -0,0 +1,214 @@
+import { useState, useEffect, useContext } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+import { deleteTodos, UpdateTodo, EditTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { AuthContext } from './Auth/AuthContext';
+
+type Props = {
+  todo: Todo,
+  foundTodoList: (u: User) => void,
+  setErrorUpdate:React.Dispatch<React.SetStateAction<boolean>>,
+  setErrorRemove: React.Dispatch<React.SetStateAction<boolean>>,
+  editTitle: React.RefObject<HTMLInputElement>,
+  setHidden: React.Dispatch<React.SetStateAction<boolean>>,
+  edditing?: Todo | null,
+  setEdditing?: any,
+};
+
+export const ToDo: React.FC<Props> = ({
+  todo,
+  foundTodoList,
+  setErrorUpdate,
+  setErrorRemove,
+  editTitle,
+  setHidden,
+  edditing,
+  setEdditing,
+}) => {
+  const user = useContext(AuthContext);
+  const [editTodoTitile, setEditTodoTitle] = useState('');
+  const [isRemovePending, setIsRemovePending] = useState<Todo | null>(null);
+  const [toggleDoubleClick, setToggleDoubleClick] = useState<Todo | null>(null);
+
+  const selectComplited = async (toDo:Todo) => {
+    try {
+      if (toDo.completed) {
+        await UpdateTodo(toDo, false);
+      } else {
+        await UpdateTodo(toDo, true);
+      }
+    } catch {
+      setErrorUpdate(true);
+      setHidden(false);
+    }
+
+    if (user) {
+      foundTodoList(user);
+    }
+  };
+
+  const removeTodo = async (toDo:Todo) => {
+    deleteTodos(toDo);
+    try {
+      setIsRemovePending(toDo);
+      await deleteTodos(toDo);
+    } catch {
+      setErrorRemove(true);
+      setHidden(false);
+      setIsRemovePending(null);
+    }
+
+    if (user) {
+      foundTodoList(user);
+    }
+  };
+
+  const editingTodo = async (toDo:Todo) => {
+    const validTodoTitle = editTodoTitile.trim();
+
+    if (validTodoTitle.length < 1) {
```

Comment:
I can add empty todo. Also, in the checklist, we have an item that redundant spaces must be deleted
![image](https://user-images.githubusercontent.com/110456763/200320322-86105287-1297-44da-872f-34bf7da939de.png)